src/fns.rs

```diff
@@ -997,6 +1108,38 @@ mod test {
         assert_eq!(distance, 4);
     }
 
+    #[test]
+    fn test_string_lessp() {
+        // Test Equality
+        assert_lisp("(string-lessp \"less\" \"less\")", "nil");
+        // Test Length
+        assert_lisp("(string-lessp \"les\" \"less\")", "t");
+        assert_lisp("(string-lessp \"less\" \"les\")", "nil");
+        // Check for less than
+        assert_lisp("(string-lessp \"abc\" \"bcd\")", "t");
+    }
+
+    #[test]
+    fn test_string_version_lessp() {
+        // Test Equality
+        assert_lisp("(string-version-lessp \"less\" \"less\")", "nil");
+        // Test Length
+        assert_lisp("(string-version-lessp \"les\" \"less\")", "t");
+        assert_lisp("(string-version-lessp \"less\" \"les\")", "nil");
+        // Test for less than
+        assert_lisp("(string-version-lessp \"abc\" \"bcd\")", "t");
+        // Test Equality with number
+        assert_lisp("(string-version-lessp \"less1\" \"less1\")", "nil");
+        // Test Differing numeric position does matter
+        assert_lisp("(string-version-lessp \"1less\" \"less1\")", "t");
+        // Test Greater than numericaly
+        assert_lisp("(string-version-lessp \"less12\" \"less1\")", "nil");
+        // Test Less than numericaly
+        assert_lisp("(string-version-lessp \"less1\" \"less12\")", "t");
+        // Test that later numbers override previous ones
+        assert_lisp("(string-version-lessp \"133less1\" \"less12\")", "t");
+    }
```

Comment:
I found a test-case the does not match Emacs

```rust
(string-version-lessp "100a" "100b")
```

Emacs returns `t`, but Rune is returning `nil`.